src/main/java/org/toradocu/extractor/JavadocExtractor.java

```diff
@@ -1,515 +1,664 @@
+      simpleName =
+          className.substring(className.lastIndexOf(".") + 1, dollarPosition + 1)
+              + clazz.getSimpleName();
+    } else {
+      // Top-level class.
+      fileName = className;
+      simpleName = clazz.getSimpleName();
+    }
+    return ImmutablePair.of(fileName, simpleName);
+  }
+
+  private List<String> getClassesInSamePackage(String className, String sourceFile) {
+    // TODO Add missing Javadoc documentation.
+    // TODO Improve the code: this method should return all the available types in a given package.
+    String packagePath = sourceFile.substring(0, sourceFile.lastIndexOf("/"));
+    File folder = new File(packagePath);
+    File[] listOfFiles = folder.listFiles();
+    List<String> classesInPackage = new ArrayList<>();
+    for (File file : listOfFiles) {
+      // This loop extracts files in the same directory as the class being analysed
+      // in order to find eventual Exception classes located in the same package.
+      // "package-info" files are not useful for this purpose.
+      String name = extractClassNameForSource(file.getName(), className);
+      if (name != null && !name.equals(className) && !name.contains("package-info")) {
+        classesInPackage.add(name);
+      }
+    }
+    return classesInPackage;
+  }
 
   /**
-   * Constructs a {@code JavadocExtractor} with the given doclet {@code configuration}.
+   * Given the file name of a source located in the same package of the Class being analysed and the
+   * name of that Class, this method compose the class name corresponding to the source.
    *
-   * @param configuration the Javadoc doclet configuration
+   * @param sourceFileName name of the source file found in package
+   * @param analyzedClassName class name of the class being analysed
+   * @return the class name of the source, or null if {@code analyzedClassName} does not contain a
+   *     dot (i.e., {@code analyzedClassName} is a simple, not-qualified name)
    */
-  public JavadocExtractor(ConfigurationImpl configuration) {
-    this.configuration = configuration;
+  private String extractClassNameForSource(String sourceFileName, String analyzedClassName) {
+    int lastDot = analyzedClassName.lastIndexOf(".");
+    if (lastDot == -1) {
+      return null;
+    }
+    return analyzedClassName.substring(0, lastDot) + "." + sourceFileName.replace(".java", "");
   }
 
   /**
-   * Returns a list of {@code DocumentedMethod}s extracted from the given {@code classDoc}.
+   * Instantiates tags (of param, return or throws kind) referred to a source member.
    *
-   * @param classDoc the {@code ClassDoc} from which to extract method documentation
-   * @return a list containing documented methods from the class
-   * @throws IOException if the method encounters an error while reading/generating class
-   *     documentation
+   * @param classesInPackage list of class names in sourceMember's package
+   * @param sourceMember the source member the tags refer to
```

Comment:
Here and elsewhere, `sourceMember` would be better named `sourceCallable` since it isn't an arbitrary member (not a field, or inner class...).  Likewise for the word "member" later in the line.